src/main/java/me/profelements/dynatech/items/electric/AntigravityBubble.java

```diff
@@ -19,16 +20,11 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.HashSet;
-import java.util.Iterator;
 import java.util.List;
-import java.util.Set;
-import java.util.UUID;
 
 public class AntigravityBubble extends AMachine {
 
-    private final Set<UUID> enabledPlayers = new HashSet<>();
```

Comment:
I'm fairly sure this is resolved here, this is a single set describing all players with the effect and it instead needs to be a Map<Location, SomeObject> that instead describes each block's enabledPlayers and then only gets and works on the relevant set.

Reply:
I don't know why I didn't think of this, I'll get working on this. This is way easier than the whole Bukkit task thing

Reply:
After some thought I think I noticed a problem with this approach which is exactly why I made the task thing; if you are in an overlapping range of 2 bubbles and fly out of range of one of the bubbles, that bubble will think you're not in "it's enabledPlayers" of it's "location" anymore and disable flight. Then a few ticks later the other antigravity bubble will grant you flight but you'd still fall to the ground.

Reply:
Best bet would be to check if the player is in another bubble at the time of attempted removal and, if they are, remove them from the original bubbles list but without setting fly to false. This can be done by running through the sets of players which could be expensive but should only occur during a transition between bubbles.

Reply:
If I understand this correctly you mean that the bubbles send their location with their enabled players to another class like how they send their location in my solution, and then when a bubble thinks a player shouldn't fly it calls a method in that class checking all the enabled players of all bubbles and disabling flight if it isn't in any of them?

This is probably less resource intensive than my current solution as it would only have to run through the sets of players if it really has to.